test/quantization/core/experimental/test_quantizer.py

```diff
@@ -1,19 +1,144 @@
+
+        # get apot quantized tensor result
+        qtensor = quantizer.quantize_APoT(tensor2quantize=tensor2quantize, use_int_repr=True)
+
+        # get uniform quantization quantized tensor result
+        uniform_quantized = quantize_per_tensor(input=tensor2quantize, scale=1.0, zero_point=0, dtype=torch.quint8).int_repr()
+
+        qtensor_data = torch.tensor(qtensor).type(torch.uint8)
+        uniform_quantized_tensor = uniform_quantized.data
+
+        self.assertTrue(torch.equal(qtensor_data, uniform_quantized_tensor))
+
+    r""" Tests quantize_APoT result (int representation) on random 2-dim tensor
+        and hardcoded values for b, k by comparing to uniform quantization
+        (non-uniform quantization reduces to uniform for k = 1)
+        quantized tensor (https://pytorch.org/docs/stable/generated/torch.quantize_per_tensor.html)
+        * tensor2quantize: Tensor
+        * b: 4
+        * k: 1
+    """
+    def test_quantize_APoT_rand_2d(self):
+        # generate random size of tensor2dequantize between 1 -> 20
+        size = random.randint(1, 20)
+
+        # generate tensor with random fp values
+        tensor2quantize = torch.rand((size, size), dtype=torch.float)
+
+        quantizer = APoTQuantizer(4, 1, False)
+
+        # get apot quantized tensor result
+        qtensor = quantizer.quantize_APoT(tensor2quantize=tensor2quantize, use_int_repr=True)
+
+        # get uniform quantization quantized tensor result
+        uniform_quantized = quantize_per_tensor(input=tensor2quantize, scale=1.0, zero_point=0, dtype=torch.quint8).int_repr()
+
+        qtensor_data = torch.tensor(qtensor).type(torch.uint8)
+        uniform_quantized_tensor = uniform_quantized.data
+
+        self.assertTrue(torch.equal(qtensor_data, uniform_quantized_tensor))
+
+    r""" Tests quantize_APoT for k != 1.
+        Tests quantize_APoT result (reduced precision fp representation) on
+        random 1-dim tensor and hardcoded values for b=4, k=2 by comparing results to
+        hand-calculated error bound (+/- 0.25 between reduced precision fp representation
+        and original input tensor values because max difference between quantization levels
+        for b=4, k=2 is 0.25).
+        * tensor2quantize: Tensor
+        * b: 4
+        * k: 2
+    """
+    def test_quantize_APoT_k2(self):
+        # generate random size of tensor2dequantize between 1 -> 20
+        size = random.randint(1, 20)
+
+        # generate tensor with random fp values
+        tensor2quantize = torch.rand((size), dtype=torch.float)
+
+        quantizer = APoTQuantizer(4, 2, False)
+
+        # get apot quantized tensor result
+        qtensor = quantizer.quantize_APoT(tensor2quantize=tensor2quantize, use_int_repr=False)
+        qtensor_data = torch.tensor(qtensor).type(torch.float)
+
+        expectedResult = True
+
+        for result, orig in zip(qtensor_data, tensor2quantize):
+            if abs(result - orig) > 0.25:
```

Comment:
there is no such guarantee actually, I think what we do typically is have a reference implementation in python and compare with that

Reply:
decided to instead test using hand-calculated example from APoT paper (https://arxiv.org/pdf/1909.13144.pdf) after talking with @dzdang